packages/hoppscotch-common/src/services/banner.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Service } from "dioc"
+import { ref } from "vue"
+import { getI18n } from "~/modules/i18n"
+
+export type BannerType = "info" | "warning" | "error"
+
+export type BannerContent = {
+  type: BannerType
+  text: (t: ReturnType<typeof getI18n>) => string
+  // Can be used to display an alternate text when display size is small
+  alternateText?: (t: ReturnType<typeof getI18n>) => string
```

Comment:
same as the last comment

```suggestion
  alternateText?: string
```